server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -229,12 +229,45 @@ func (e *EventChannel) HandleEvent(ctx context.Context, event *pepb.PublishBuild
 	// something to show the user. Flushing the proto file here allows that when the
 	// client fetches status for the incomplete build. Also flush if we haven't in over a minute.
 	if isWorkspaceStatusEvent(bazelBuildEvent) || e.pw.TimeSinceLastWrite().Minutes() > 1 {
-		return e.pw.Flush(ctx)
+		if err := e.pw.Flush(ctx); err != nil {
+			return err
+		}
+	}
+	// When we get the workspace status event, update the invocation in the DB
+	// so that it can be searched by its commit SHA, user name, etc. even
+	// while the invocation is still in progress.
+	if isWorkspaceStatusEvent(bazelBuildEvent) {
+		if err := e.writeBuildMetadata(ctx, iid); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *EventChannel) writeBuildMetadata(ctx context.Context, invocationID string) error {
+	db := e.env.GetInvocationDB()
+	events, err := e.readAllTempBlobs(ctx, invocationID)
+	if err != nil {
+		return err
+	}
+	// TODO: Make sure the lookup/insert happens in a single transaction.
+	// This isn't an issue currently since all invocation-modifying events
+	// are coming through the PublishBuildEvent stream, but using a transaction
+	// here would make this more future-proof.
+	ti, err := db.LookupInvocation(ctx, invocationID)
```

Comment:
do we even need to do a lookup here? we don't do that when finalizing in FinalizeInvocation

Reply:
We don't really need to -- changed it so it doesn't do the lookup. I was just a bit concerned that we're depending on GORM's hidden-ish behavior of not updating fields that are falsy (if it didn't have that behavior then we could potentially be deleting fields that we wrote in the build started event). But if we're not doing a lookup in FinalizeInvocation either, we may as well not do it here